File: secureflow/src/lib/businessEmail.ts
/**
 * SecureFlow is a workplace product: accounts must be tied to a company domain.
 * Anything not on this personal-provider denylist is treated as a business domain.
 */
const PERSONAL_EMAIL_DOMAINS = new Set([
  "gmail.com",
  "googlemail.com",
  "outlook.com",
  "hotmail.com",
  "live.com",
  "msn.com",
  "yahoo.com",
  "yahoo.co.in",
  "yahoo.co.uk",
  "ymail.com",
  "rocketmail.com",
  "icloud.com",
  "me.com",
  "mac.com",
  "aol.com",
  "protonmail.com",
  "proton.me",
  "pm.me",
  "gmx.com",
  "gmx.net",
  "mail.com",
  "yandex.com",
  "yandex.ru",
  "zoho.com",
  "zohomail.com",
  "rediffmail.com",
  "inbox.com",
  "fastmail.com",
  "tutanota.com",
  "tuta.io",
  "mailinator.com",
  "10minutemail.com",
  "guerrillamail.com",
  "temp-mail.org",
]);

export const BUSINESS_EMAIL_FIELD_ERROR = "Please use your business email address";

export const BUSINESS_EMAIL_BLOCKED_MESSAGE =
  "SecureFlow requires a business email address — personal email accounts (Gmail, Yahoo, Outlook, etc.) aren't supported. Please sign in with your work account.";

export function isBusinessEmail(email: string): boolean {
  const domain = email.trim().toLowerCase().split("@")[1];
  if (!domain) return false;
  return !PERSONAL_EMAIL_DOMAINS.has(domain);
}

File: backend/utils/business_email.py
"""
utils/business_email.py

SecureFlow only allows sign-in from business/company email addresses —
free personal email providers are rejected. This is a denylist approach
(block known personal providers, allow everything else) rather than an
allowlist, since we don't know every company's domain in advance and a
custom domain is itself decent evidence of a business account.

This is NOT foolproof — someone could register yet another free provider
we haven't listed, or a domain we've blocked could actually be used by a
business (e.g. some companies do use a Zoho/GMX address). Treat this as a
practical filter, not a strong identity guarantee.
"""

PERSONAL_EMAIL_DOMAINS = {
    # Google
    "gmail.com", "googlemail.com",
    # Microsoft
    "outlook.com", "hotmail.com", "live.com", "msn.com",
    # Yahoo
    "yahoo.com", "yahoo.co.in", "yahoo.co.uk", "ymail.com", "rocketmail.com",
    # Apple
    "icloud.com", "me.com", "mac.com",
    # Other common free/personal providers
    "aol.com", "protonmail.com", "proton.me", "pm.me",
    "gmx.com", "gmx.net", "mail.com", "yandex.com", "yandex.ru",
    "zoho.com", "zohomail.com",
    "rediffmail.com", "inbox.com", "fastmail.com",
    "tutanota.com", "tuta.io",
    # Disposable/throwaway providers
    "mailinator.com", "10minutemail.com", "guerrillamail.com", "temp-mail.org",
}


def is_business_email(email: str | None) -> bool:
    """True if `email`'s domain is NOT a known personal/free provider.
    Returns False (rejected) for missing/malformed email too."""
    if not email or "@" not in email:
        return False
    domain = email.rsplit("@", 1)[-1].strip().lower()
    return domain not in PERSONAL_EMAIL_DOMAINS
